Make footer sticky for screen sizes larger than 865px height

kennethlimjf opened the pull request.

tgxworld commented:
@kennethlimjf Thanks for contributing!! However, I'm not too sure what we are trying to fix here. Could you provide some screenshots of the before and after?

yangshun commented:
It's so that the footer will be at the bottom of the screen even though the site content isn't taller than the screen height. NUSMods does it too.

It's a good idea! But `flex` is not supported by IE9. @tgxworld do you want to support IE9? Haha.

Alternatively, there's a non-`flex` solution but involves hardcoding of the footer height.

yangshun commented:
I actually don't see a page that has so little content that the footer isn't at the bottom. Perhaps there isn't a need for a sticky footer.

yangshun commented on app/assets/stylesheets/base.scss:
```diff
@@ -1,3 +1,13 @@
   background-color: white;
 }
+
+.site {
```
I feel that adding this styles directly to `body` is ok, no need for a `.site` class.

tgxworld commented:
> It's so that the footer will be at the bottom of the screen even though the site content isn't taller than the screen height. NUSMods does it too.
> 
> It's a good idea! But flex is not supported by IE9. @tgxworld do you want to support IE9? Haha.
> 
> Alternatively, there's a non-flex solution but involves hardcoding of the footer height.

![screenshot from 2015-06-13 00 16 22](https://cloud.githubusercontent.com/assets/4335742/8134394/72e408d8-1161-11e5-983d-72dc0e708608.png)

Ahh Ic. Lets move the web forward by dropping IE9 :trollface:

kennethlimjf commented:
Ah ya should have added a screenshot. I added the sticky footer since it appeared non-sticky on my 23" monitor. But for most laptop screens, it shouldn't be a problem.

But anyways, now it should look good on all screens with this PR :smile: (with the exception of IE9)

Updated the CSS not to use class.

tgxworld commented:
@yangshun I'll let you merge this if its ok :smile:

yangshun commented:
Just tried it. It works really well and elegant. Thanks!

tgxworld commented:
@kennethlimjf This is awesome thanks!

kennethlimjf commented:
Thanks :)

On Tue, Jun 16, 2015 at 12:56 AM, Guo Xiang Tan <notifications@github.com>
wrote:

> @kennethlimjf <https://github.com/kennethlimjf> This is awesome thanks!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ruby-bench/ruby-bench-web/pull/89#issuecomment-112135815>
> .
>
